src/main/kotlin/com/cognifide/gradle/aem/pkg/PackagePlugin.kt

```diff
@@ -62,23 +62,27 @@ class PackagePlugin : CommonDefaultPlugin() {
                 }
             }
         }
-        val upload = register<PackageUpload>(PackageUpload.NAME) {
+        val validate = register<PackageValidate>(PackageValidate.NAME) {
             dependsOn(compose)
+            packages.from(compose.flatMap { it.archiveFile })
+        }
+        val upload = register<PackageUpload>(PackageUpload.NAME) {
+            dependsOn(compose, validate)
```

Comment:
I’m on the fence about needing to validate before upload. On one hand validating costs time, on the other, searching for issues manually if something doesn’t work after the upload also costs time. Maybe we could validate in parallel to uploading?

Reply:
I am too, but ... generally, validation takes most often < 3 s, but failed upload and looking for error could take the same time or be greater. and also we could avoid breaking instance repository consistency (state `package installed with errors`).

if you want to play without validation simply add `package.validator.enabled=false` to _gradle.properties_ file.

the main goal of GAP is to be the smartest as it is possible leveraging build speed.
it should take care of mostly everything without configuration. the case you mentioned is rather a performance improvement but reduces safety. if you want to have a quicker build you should work without validation then run validation finally but separately before final commit/relase/deploy etc.